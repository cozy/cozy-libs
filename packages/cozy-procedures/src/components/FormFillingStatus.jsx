import React from 'react'
import { translate } from 'cozy-ui/transpiled/react/I18n'

export const FormFillingStatus = ({ completed, total, t }) => {
  if (completed === '0') {
    return null
  }

  return (
    <div>
      <span className="u-weirdGreen">
        {t('personalDataForm.completedStatus.main', { completed, total })}
      </span>{' '}
      {t('personalDataForm.completedStatus.rest')}
    </div>
  )
}

export default translate()(FormFillingStatus)
