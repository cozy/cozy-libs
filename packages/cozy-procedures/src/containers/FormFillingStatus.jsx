import { connect } from 'react-redux'

import context from '../redux/context'
import FormFillingStatus from '../components/FormFillingStatus'
import { getCompletedFields, getTotalFields } from '../redux/personalDataSlice'

export const mapStateToProps = state => ({
  completed: getCompletedFields(state),
  total: getTotalFields(state)
})

export default connect(
  mapStateToProps,
  null,
  null,
  { context }
)(FormFillingStatus)
