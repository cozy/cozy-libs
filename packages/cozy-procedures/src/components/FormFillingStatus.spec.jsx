import React from 'react'

import { FormFillingStatus } from './FormFillingStatus'

const fakeT = (msgId, args) => {
  return `${msgId} ${JSON.stringify(args)}`
}

describe('FormFillingStatus component', () => {
  it('should display nothing if no data is filled', () => {
    const component = shallow(
      <FormFillingStatus completed="0" total="10" t={fakeT} />
    ).getElement()
    expect(component).toBeNull()
  })

  it('should display a status if there are some completed data', () => {
    const component = mount(
      <FormFillingStatus completed="5" total="10" t={fakeT} />
    )
    expect(component).toMatchSnapshot()
  })
})
