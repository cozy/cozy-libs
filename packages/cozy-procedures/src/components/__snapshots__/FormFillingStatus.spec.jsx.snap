// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFillingStatus component should display a status if there are some completed data 1`] = `
<FormFillingStatus
  completed="5"
  t={[Function]}
  total="10"
>
  <div>
    <span
      className="u-weirdGreen"
    >
      personalDataForm.completedStatus.main {"completed":"5","total":"10"}
    </span>
     
    personalDataForm.completedStatus.rest undefined
  </div>
</FormFillingStatus>
`;
